source/game/system/map/MapdataCheckPoint.hh

```diff
@@ -28,12 +28,34 @@ public:
     };
     STATIC_ASSERT(sizeof(SData) == 0x14);
 
+    enum class SectorOccupancy {
+        InsideSector,               ///< Player is inside the given checkpoint group
+        OutsideSector,              ///< Player is outside the given checkpoint group
+        OutsideSector_BetweenSides, ///< Player is between sides of the checkpoint group but not
+                                    ///< between this checkpoint and next
+    };
+
     MapdataCheckPoint(const SData *data);
     void read(EGG::Stream &stream);
     void initCheckpointLinks(MapdataCheckPointAccessor &accessor, int id);
+    [[nodiscard]] SectorOccupancy checkSectorAndDistanceRatio(const EGG::Vector3f &pos,
+            f32 &distanceRatio) const;
+
+    /// @beginSetters
+    void setSearched();
+    void clearSearched();
+    /// @endSetters
 
     /// @beginGetters
+    [[nodiscard]] bool searched() const;
     [[nodiscard]] s8 checkArea() const;
+    [[nodiscard]] bool isNormalCheckpoint() const;
+    [[nodiscard]] bool isFinishLine() const;
```

Comment:
These two aren't necessarily getters, they're just helper functions to get more information about the checkpoint. They should not be grouped together with raw member getters.